implementations/rust/ockam/ockam_command/src/identity/show.rs

```diff
@@ -32,7 +35,10 @@ fn run_impl(opts: CommandGlobalOpts, cmd: ShowCommand) -> crate::Result<()> {
         );
     }
     let state = opts.state.identities.get(&cmd.name)?;
-    if cmd.full {
+    if cmd.hex {
+        let identity = state.config.change_history.export()?;
+        print_output(identity, &opts.global_args.output_format)?;
+    } else if cmd.full {
```

Comment:
This is something I've been thinking about trying to support globally as well.

Perhaps we have a flag similar to `--output json` but instead `--encoding hex`  (which would allow for base64 use cases in the future etc)

Reply:
Does not needed to be fully added in this PR, maybe just change this command and examples to accept encoding.

Write a `// TODO `

And I can update it to be a global option in my coming PR.

Reply:
@CAOakleyII why not `--output hex`?

Reply:
replaced `--hex`  with `--encoding hex`   (only applicable if `--full` was specified) and added a TODO to review if that would be something general on other commands   as well.  The only other that I think right now could  make sense is for credential show.   But even there perhaps settling on some json format is more user friendly).

Reply:
> @CAOakleyII why not --output hex?

@etorreborre 
I thought about that at first.

Few reasons, some very opinionated and could be wrong - 

* I view output primarily as formatting, human readable but machine serializable outputs. (JSON, YAML, TOML). 

* They are technically mutually exclusive, you could technically have a base64/hex encoded json output.

* _Slightly_ more refactoring required, our current `--output` defines that the struct has traits of `Output` and `serde::Serialize` where as a hex encoding does not need that.